Dockerfile

```diff
@@ -37,12 +37,11 @@ RUN case ${TARGETPLATFORM} in \
     GOARCH=${GOARCH} VERSION=${VERSION} make build && touch jwt_signing_key.pem
 
 # Copy binary to alpine
-FROM alpine:3.15
+FROM gcr.io/distroless/static-debian11:nonroot
 COPY nsswitch.conf /etc/nsswitch.conf
-COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
 COPY --from=builder /go/src/github.com/oauth2-proxy/oauth2-proxy/oauth2-proxy /bin/oauth2-proxy
 COPY --from=builder /go/src/github.com/oauth2-proxy/oauth2-proxy/jwt_signing_key.pem /etc/ssl/private/jwt_signing_key.pem
 
-USER 2000:2000
+USER nonroot
```

Comment:
What's the difference between setting this as `nonroot` and the unknown 2000 user?

Reply:
I think that `nonroot` is an actual user in the base image.

Reply:
`nonroot` has the `uid/gid` `65532`
```
root:x:0:0:root:/root:/sbin/nologin
nobody:x:65534:65534:nobody:/nonexistent:/sbin/nologin
nonroot:x:65532:65532:nonroot:/home/nonroot:/sbin/nologin
```

Reply:
A benefit of using the `nonroot`, it's easy to find information about that specific uid. First result on my search results, links to the Distroless repo.

Reply:
I replaced `nonroot` with the matching uid/gid, since Alpine does not have that user entry, but it can still start with an unkown uid/gid.